pkg/api/defaults-kubelet.go

```diff
@@ -163,6 +163,18 @@ func (cs *ContainerService) setKubeletConfig(isUpgrade bool) {
 		setMissingKubeletValues(cs.Properties.MasterProfile.KubernetesConfig, o.KubernetesConfig.KubeletConfig)
 		addDefaultFeatureGates(cs.Properties.MasterProfile.KubernetesConfig.KubeletConfig, o.OrchestratorVersion, "", "")
 
+		if isUpgrade && common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.14.0") {
+			hasSupportPodPidsLimitFeatureGate := strings.Contains(o.KubernetesConfig.KubeletConfig["--feature-gates"], "SupportPodPidsLimit=true")
```

Comment:
This should get the --feature-gates flag from the MasterProfile

Reply:
Yes, My original intention was to use the feature gates flag from the MasterProfile. On the other hand, using the feature gates from the OrchestratorProfile will avoid any bug that the feature gate flag in MasterProfile is not correctly set. Anyway, I will use the feature gates flag from the MasterProfile.